lib/internal/Magento/Framework/App/DeploymentConfig.php

```diff
@@ -140,7 +140,7 @@ public function isDbAvailable()
      */
     private function load()
     {
-        if (null === $this->data) {
+        if (!is_array($this->data)) {
```

Comment:
I have doubts regarding this change. Basically, we need to check that the `$data` has been initialized with some data. Prior to that point `$data` is `null`. Technically `is_array` might be slower that strict comparison with `null`. So we don't need to introduce changes that may affect performance without necessity.

Reply:
I agree, this change is not needed. My bad. I will double check the rest of the changes. I will try to remove data reset from \Magento\Framework\App\DeploymentConfig::isAvailable also if possible.